### PR TITLE
glib: Add connect_notify* methods to SignalGroup

### PR DESCRIPTION
**Pull request by melix99**
Fixes https://github.com/gtk-rs/gtk-rs-core/issues/1021.

Didn't have a chance to test it locally because I couldn't use cargo patch in my project :/

### CODE REVIEW

File: glib/src/gobject/signal_group.rs [hunk 1]
```diff
@@ -42,6 +42,44 @@ impl SignalGroup {
         self.connect_closure(signal_name, after, RustClosure::new_local(callback));
     }
 
+    #[inline]
+    pub fn connect_notify<F>(&self, name: Option<&str>, callback: F)
+    where
+        F: Fn() + Send + Sync + 'static,
```
Comment (sdroege) [inline glib/src/gobject/signal_group.rs:48]: You probably want to pass the object and pspec into the closure

File: glib/src/gobject/signal_group.rs [hunk 2]
```diff
@@ -42,6 +42,44 @@ impl SignalGroup {
         self.connect_closure(signal_name, after, RustClosure::new_local(callback));
     }
 
+    #[inline]
+    pub fn connect_notify<F>(&self, name: Option<&str>, callback: F)
```
Comment (sdroege) [inline glib/src/gobject/signal_group.rs:46]: Why is this needed? It should be possibly to directly call http://gtk-rs.org/gtk-rs-core/stable/latest/docs/glib/object/trait.ObjectExt.html#tymethod.connect_notify on the `SignalGroup` as it's a `glib::Object`

Comment (melix99) [reply glib/src/gobject/signal_group.rs:46]: Because of the nature of `SignalGroup`.

`SignalGroup` is an object that makes it easy to listen to multiple signals from the `target` object. This means that the `SignalGroup::connect` will listen for signals of the `target` object, instead of the `SignalGroup` itself. This PR adds an helper to connect to notify signals of properties of the `target` object, which is different than `ObjectExt::connect_notify`, because the latter only listens to notify signals of the `SignalGroup` object, which is not what we want.

File: glib/src/gobject/signal_group.rs (hunk 1 above)
Comment (melix99) [reply glib/src/gobject/signal_group.rs:48]: I'm not sure how to get the pspec of the `target` object though.

Comment (bilelmoussaoui) [reply glib/src/gobject/signal_group.rs:48]: Using https://gtk-rs.org/gtk-rs-core/stable/latest/docs/glib/object/trait.ObjectExt.html#tymethod.find_property?

Comment (sdroege) [reply glib/src/gobject/signal_group.rs:48]: It's passed as argument to the callback

Comment (sdroege) [reply glib/src/gobject/signal_group.rs:48]: Together with the object itself

File: glib/src/gobject/signal_group.rs [hunk 3]
```diff
@@ -42,6 +42,50 @@ impl SignalGroup {
...
+        F: Fn(&Self, &crate::ParamSpec) + Send + Sync + 'static,
+    {
+        let signal_name = if let Some(name) = name {
+            format!("notify::{name}")
+        } else {
+            "notify".into()
+        };
+
+        let closure = crate::RustClosure::new(move |values| {
+            let obj = values[0].get().unwrap();
```
Comment (melix99) [inline glib/src/gobject/signal_group.rs:57]: Ah, it returns the target object instead of the SignalGroup. Not sure what to do here. I think we can either return a glib::Object of the target object or clone the SignalGroup and return that instead. I'm inclined for the former.

Comment (melix99) [reply glib/src/gobject/signal_group.rs:57]: I'll push that for now.

Comment (sdroege) [reply glib/src/gobject/signal_group.rs:57]: Yes that's what I would expect. The other signals should work the same way

**Review by sdroege — APPROVED** [review]
